### PR TITLE
IterativeVertexFinder: fill vertex-to-particle relation

### PR DESCRIPTION
**Pull request by starsdong**
### Briefly, what does this PR introduce?
- added association for ReconstructionParticles to output vertices

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [X] New feature (issue #1575)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [X] Tests for the changes have been added
- [X] Documentation has been added / updated
- [X] Changes have been communicated to collaborators

### Does this PR change default behavior?
- ReconstructedParticle association added to the default output CKFCentralVertices

### CODE REVIEW

File: src/global/tracking/IterativeVertexFinder_factory.h
Comment (veprbl) [inline src/global/tracking/IterativeVertexFinder_factory.h:30]:
```suggestion
    PodioInput<edm4eic::ReconstructedParticle> m_edm4eic_reconParticles_input {this};
```
Not sure how a `const char*` converts to `InputOptions`, but it seems like this second argument is not even needed.

File: src/algorithms/tracking/IterativeVertexFinder.cc
Comment (veprbl) [inline src/algorithms/tracking/IterativeVertexFinder.cc:41]:
```suggestion
#include <edm4eic/Trajectory.h>
#include <edm4eic/unit_system.h>
```

File: src/algorithms/tracking/IterativeVertexFinder.cc
Comment (veprbl) [inline src/algorithms/tracking/IterativeVertexFinder.cc:241]:
```suggestion
            constexpr acts_to_edm4eic = edm4eic::unit::mm / Acts::UnitConstants::mm;
            if(fabs(par.getLoc().a - loc_a * acts_to_edm4eic) < 1.e-4 && fabs(par.getLoc().b - loc_b * acts_to_edm4eic) < 1.e-4) {
```

File: src/global/tracking/tracking.cc
Comment (starsdong) [reply src/global/tracking/tracking.cc:229]: Dmitri, we have shown the realistic seeded trajectories deliver good performance. And I believe we agreed in the collaboration to move to the Seeded objects for these downstream analysis.

File: src/global/tracking/IterativeVertexFinder_factory.h
Comment (starsdong) [inline src/global/tracking/IterativeVertexFinder_factory.h:1]: Dmitry, this changes break the compilation. I guess your proposal may work, but that would require also changes in the IterativeVertexFinder.cc to make the corresponding changes. For simplicity, shall we revert the commit and keep the original version in the PR?

File: src/algorithms/tracking/IterativeVertexFinder.cc
Comment (wdconinc) [inline src/algorithms/tracking/IterativeVertexFinder.cc:69]: This is what needs changing to fix your compilation. We should not introduce new interfaces that use `std::vecotr<podio_type>`.

**Review by wdconinc — CHANGES_REQUESTED** [review]

File: src/global/tracking/tracking.cc
Comment (veprbl) [reply src/global/tracking/tracking.cc:229]: Yes, but we should move centrally. The plan is to rename `ReconstructedSeededChargedParticles` to `ReconstructedChargedParticles` and rename the original `ReconstructedChargedParticles` to something else, and leave that for dedicated tracking studies.

File: src/algorithms/tracking/IterativeVertexFinder.cc
Comment (starsdong) [reply src/algorithms/tracking/IterativeVertexFinder.cc]: addressed in the following commit

**Comment by veprbl** [comment]
image
Associations are filled.

**Review by veprbl — DISMISSED** [review]
LGTM

**Comment by starsdong** [comment]
Hi Woulter and Dmitry,

Are these anything you would like to comment before merging this PR? The requested change Woulter brought up, I believe, is addressed in the later commits. Would appreciate if you can take a review to see whether we can proceed. Thanks

/xin

**Comment by veprbl** [comment]
I don't understand why this hasn't merged over the weekend. Maybe we needed to have and approval form @wdconinc or dismiss his "1 change requested" review.

File: src/algorithms/tracking/IterativeVertexFinder.cc
Comment (wdconinc) [inline src/algorithms/tracking/IterativeVertexFinder.cc:244]: Avoid dimensionful magic numbers.

**Review by wdconinc — APPROVED** [review]
The 'request changes' is always sticky and requires the same person who requested changes to give explicit approval. In this case, I requested changes since we needed to use the podio collection instead of std::vector<podio type>. That's been done now, so I approve.

One more comment though on a magic number that should be a configuration variable (best) or defined as a ratio of another dimensionful quantity. The reconstruction should not care whether we work in dd4hep units or, for some reason, another unit system.

File: src/algorithms/tracking/IterativeVertexFinder.cc
Comment (veprbl) [reply src/algorithms/tracking/IterativeVertexFinder.cc]: Addressed. Let's assume that we don't need to change this value. The algorithm depends on numerical tolerance, and it will be replaced with another approach once that's supported by the EDM.

File: src/algorithms/tracking/IterativeVertexFinder.cc
Comment (wdconinc) [inline src/algorithms/tracking/IterativeVertexFinder.cc:229]: Fails to compile for acts v33, with
```
#32 1016.3 /tmp/root/spack-stage/spack-stage-eicrecon-git.7f8feba7572a7b05cab02027929cd08659b1cdbd_main-xib6spjd5y4zv24smqpnu3i6uxgmhggo/spack-src/src/algorithms/tracking/IterativeVertexFinder.cc:229:25: error: no matching function for call to object of type 'InputTrack::Extractor' (aka 'Delegate<GenericBoundTrackParameters<ParticleHypothesis> (const Acts::InputTrack &)>')
#32 1016.3       const auto& par = finderCfg.extractParameters(trk);
#32 1016.3                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
#32 1016.3 /opt/software/linux-debian12-x86_64_v2/clang-16.0.6/acts-33.1.0-eihfjd5ywurf5qidbrexfxoreo7yj4nx/include/Acts/Utilities/Delegate.hpp:249:15: note: candidate function not viable: no known conversion from 'const InputTrack *const' to 'const Acts::InputTrack' for 1st argument; dereference the argument with *
#32 1016.3   return_type operator()(Args... args) const {
#32 1016.3               ^
#32 1016.3 1 error generated.
#32 1016.3 make[3]: *** [src/algorithms/tracking/CMakeFiles/algorithms_tracking_library.dir/build.make:163: src/algorithms/tracking/CMakeFiles/algorithms_tracking_library.dir/IterativeVertexFinder.cc.o] Error 1
```

Comment (wdconinc) [reply src/algorithms/tracking/IterativeVertexFinder.cc:229]: Ref https://eicweb.phy.anl.gov/containers/eic_container/-/jobs/3746414

Comment (wdconinc) [reply src/algorithms/tracking/IterativeVertexFinder.cc:229]: #1610

**Pull request merged**
